### PR TITLE
Books

### PR DESCRIPTION
**Pull request by sdandois**
New Components
------------------

BooksList: Display flat list of BooksItems
BookItem: DIsplay info of a book title, auhor and book cover if exists. If not show placeholder image

New Actions for requesting the data from API.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+module.exports = {
```
Comment (fbazzana) [inline .eslintrc.js:1]: Is necessary this ```.js``` extension?

File: src/BooksScreen.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React, { Component } from 'react';
+import { View } from 'react-native';
+
+import BookList from './components/BookList';
+import styles from './styles/booksScreen';
+
+class BooksScreen extends Component {
+  static navigationOptions = {
```
Comment (fbazzana) [inline src/BooksScreen.js]: Define this titles in another file

File: src/actions/types.js [hunk 3]
```diff
@@ -3,3 +3,6 @@ export const ADD_TODO = 'add_todo';
 export const TOGGLE_DONE = 'toggle_done';
 export const REMOVE_TODO = 'remove_todo';
 export const REMOVE_COMPLETED = 'remove_completed';
+export const REQUEST_BOOKS_START = 'request-books-start';
+export const REQUEST_BOOKS_OK = 'request-books-ok';
+export const REQUEST_BOOKS_FAIL = 'request-books-fail';
```
Comment (fbazzana) [inline src/actions/types.js]: Let's change this to our actions naming convention. Use ```snakecase (request_books)``` and ```uppercase (REQUEST_BOOKS)``` in actions. Also, the 'ok' actions should have ```_SUCCESS``` suffix, and 'failed' actions should have ```_FAILURE```.

Like this:
```REQUEST_BOOKS```
```REQUEST_BOOKS_SUCCESS```
```REQUEST_BOOKS_FAILURE```

File: src/components/BookList.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import * as statusCodes from '../reducers/constants';
+
+import styles from './styles/bookList';
+import BookItem from './BookItem';
+import { Spinner } from './common';
+
+class BookList extends Component {
+  componentWillMount() {
+    this.props.requestBooks();
+  }
```
Comment (fbazzana) [inline src/components/BookList.js:17]: Only as a comment, ```componentWillMount``` will be deprecated in future ```React``` versions. You could replace it for ```componentDidMount```.

File: src/components/common/styles/spinner.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { StyleSheet } from 'react-native';
```
Comment (fbazzana) [inline src/components/common/styles/spinner.js:1]: Define this inside the component's folder

File: src/components/styles/bookItem.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { StyleSheet } from 'react-native';
```
Comment (fbazzana) [inline src/components/styles/bookItem.js:1]: Same here

**Review by fbazzana — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (sdandois) [reply .eslintrc.js:1]: Yes. Whithout the extension is interpreted as YAML and I get this error:
```
YAMLException: Cannot read config file: /Users/widergydev/code/training/.eslintrc
Error: end of the stream or a document separator is expected at line 2, column 8:
      "env": {
```

File: src/components/BookList.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+      return <this.List />;
+    }
+    return <this.ErrorComponent />;
+  }
+}
+
+const mapStateToProps = state => ({
+  books: state.booksStore.data.map(book => {
+    if (!book.thumbnail) {
+      return { ...book, thumbnail: require('./img/descarga.jpeg') };
```
Comment (redenwidergy) [inline src/components/BookList.js]: Try importing this with import at the top of your file:
import _imageName_ from './img/descarga.jpeg';

File: src/components/BookList.js (hunk 4 above)
Comment (redenwidergy) [reply src/components/BookList.js:17]: I agree with this review :+1:

**Review by redenwidergy — APPROVED** [review]
Finish with @fbazzana's requested changes and you'll be all done!

File: src/components/common/styles/spinner.js (hunk 5 above)
Comment (sdandois) [reply src/components/common/styles/spinner.js:1]: See #6

File: src/components/styles/bookItem.js (hunk 6 above)
Comment (sdandois) [reply src/components/styles/bookItem.js:1]: See #6

**Pull request merged**